File: fix_alpha.js
const Jimp = require('jimp');

async function fixAlpha() {
    console.log('Reading images...');
    const orig = await Jimp.read('assets/rocket_banner.png');
    const clean = await Jimp.read('assets/rocket_banner_clean.png');
    
    console.log('Resizing clean image if necessary...');
    if (orig.bitmap.width !== clean.bitmap.width || orig.bitmap.height !== clean.bitmap.height) {
        clean.resize(orig.bitmap.width, orig.bitmap.height);
    }
    
    console.log('Transferring alpha channel...');
    orig.scan(0, 0, orig.bitmap.width, orig.bitmap.height, function(x, y, idx) {
        // orig.bitmap.data is a Buffer
        // idx+3 is alpha
        clean.bitmap.data[idx + 3] = orig.bitmap.data[idx + 3];
    });
    
    console.log('Saving result...');
    await clean.writeAsync('assets/rocket_banner_final.png');
    console.log('Success');
}

fixAlpha().catch(console.error);

File: fix_alpha.py
from PIL import Image

try:
    img_orig = Image.open('assets/rocket_banner.png').convert("RGBA")
    img_clean = Image.open('assets/rocket_banner_clean.png').convert("RGBA")
    
    # Resize clean image to match original if generative AI changed the dimensions
    if img_orig.size != img_clean.size:
        img_clean = img_clean.resize(img_orig.size, Image.Resampling.LANCZOS)

    # Extract alpha from original
    r, g, b, a = img_orig.split()
    
    # Extract rgb from clean
    cr, cg, cb, ca = img_clean.split()
    
    # Merge clean rgb with original alpha
    final_img = Image.merge("RGBA", (cr, cg, cb, a))
    final_img.save('assets/rocket_banner_final.png')
    print("Success")
except Exception as e:
    print(f"Error: {e}")
